From fd82384acc0405ead38ea0d9712c9a1b57913c35 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Sun, 22 Sep 2019 10:57:02 +0800
Subject: [PATCH] fips: continuing without Jitter RNG

Continue without Jitter RNG for fips to workaround alg self-tests failure
...
[    0.311313] alg: ecdh: test failed on vector 2, err=-14
[    0.311898] Kernel panic - not syncing: alg: self-tests for ecdh-generic (ecdh) failed in fips mode!
...

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 crypto/drbg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crypto/drbg.c b/crypto/drbg.c
index b6929eb..d677da5 100644
--- a/crypto/drbg.c
+++ b/crypto/drbg.c
@@ -1577,7 +1577,7 @@ static int drbg_instantiate(struct drbg_state *drbg, struct drbg_string *pers,
 		if (IS_ERR(drbg->jent)) {
 			ret = PTR_ERR(drbg->jent);
 			drbg->jent = NULL;
-			if (fips_enabled || ret != -ENOENT)
+			if (ret != -ENOENT)
 				goto free_everything;
 			pr_info("DRBG: Continuing without Jitter RNG\n");
 		}
-- 
2.7.4

